packages/forklift-console-plugin/src/modules/Providers/ProviderRow.tsx

```diff
@@ -30,76 +38,51 @@ interface CellProps {
   currentNamespace?: string;
 }
 
-/**
- * assumes that if condition is 'True' then
- * this is a positive state (success, "green")
- * i.e. ConnectionTestSucceeded
- */
-const toPositiveState = (conditionValue: string): 'Error' | 'Ok' | 'Unknown' => {
-  switch (conditionValue) {
-    case 'True':
-      return 'Ok';
-    case 'False':
-      return 'Error';
-    default:
-      return 'Unknown';
-  }
+const statusIcons = {
+  ValidationFailed: <RedExclamationCircleIcon />,
```

Comment:
1. the move away from legacy code is OK but it deserves a separate PR (it's not connected with jsonpath)
2. the icons  from `@openshift-console/dynamic-plugin-sdk` complicate unit testing (they work only because of mocks/shim layer) and can be easily replaced with PF4